encoding/encoding.go

```diff
@@ -3,4 +3,5 @@ package encoding
 // For allocating a buffer large enough to hold uint64 values.
 const (
 	uint64SizeBytes = 8
+	uint64NumBits   = 64
```

Comment:
nit: `uint64SizeBits = uint64SizeBytes * 8` to be consistent? Also for template code perhaps just embed the const `64` since otherwise you'll need to define this const eveywhere where the template is instantiated.